status-client-spec.md

```diff
@@ -109,10 +125,43 @@ v5](https://github.com/ethereum/devp2p/blob/master/discv5/discv5.md) and
 [modifications](https://github.com/status-im/rendezvous#differences-with-original-rendezvous)).
 Additionally, some static nodes MAY also be used.
 
-This part of the system is currently underspecified and requires further detail.
+A Status client MUST use at least one discovery method or use static nodes
+to communicate with other clients.
+
+Discovery V5 uses bootstrap nodes to discover other peers. Bootstrap nodes MUST support
+Discovery V5 protocol as well in order to provide peers. It is kademlia-based discovery mechanism
+and it might consume significant (at least on mobile) amount of network traffic to operate.
+
+In order to take advantage from simpler and more mobile-friendly peers discovery mechanism,
+i.e. Rendezvous protocol, one MUST provide a list of Rendezvous nodes which speak
+Rendezvous protocol. Rendezvous protocol is request-response discovery mechanism.
+It uses Ethereum Node Records (ENR) to report discovered peers.
+
+Both peers discovery mechanisms use topics to provide peers with certain capabilities.
+There is no point in returning peers that do not support a particular protocol.
+Status nodes that want to be discovered MUST register to Discovery V5 and/or Rendezvous
+with the `whisper` topic. Status nodes that are mail servers and want to
+be discoverable MUST additionally register with the `whispermail` topic.
```

Comment:
How does mailserver discovery work? I thought this was just as a normal Whisper node and that "discovery" was a manual process of adding static Whisper mailserver nodes

Reply:
Mail servers can be discovered separately from Whisper nodes. Not all Whisper nodes are mail servers so if you'd like to discover only mail servers it should be possible. That's why we have separate topics.

A mail server node would usually register itself using both `whisper` and `whipermail` topics in the peers discovery protocol.

> I thought this was just as a normal Whisper node and that "discovery" was a manual process of adding static Whisper mailserver nodes

It works this way in our client now but there is nothing against discovering mail servers through the peers discovery algorithm. The problem is that you don't know whether you can trust these discovered mail servers. There was some work around that. cc @cammellos

Reply:
> whisper and whipermail topics in the peers discovery protocol.

Ack

---

Can we add these details to the spec? Right now this is largely implicit knowledge